app/controllers/articles_controller.rb

```diff
@@ -189,6 +192,18 @@ def base_editor_assigments
     @organizations = @user&.organizations
     @tag = Tag.find_by(name: params[:template])
     @prefill = params[:prefill].to_s.gsub("\\n ", "\n").gsub("\\n", "\n")
+    set_user_approved_liquid_tags
+  end
+
+  def set_user_approved_liquid_tags
+    @user_approved_liquid_tags =
+      if @user
+        RESTRICTED_LIQUID_TAGS.filter_map do |liquid_tag|
+          liquid_tag if liquid_tag::VALID_ROLES.any? { |role| @user.has_role?(*Array(role)) }
+        end
+      else
+        []
+      end
```

Comment:
This is a sort of quick patch so that, in the view, we can check if the user is authorized to use a restricted liquid tag before showing the documentation for it. It's flexible enough that all we have to do for a future restricted tag is add it to the `RESTRICTED_LIQUID_TAGS` constant.

Ideally, this logic probably wouldn't live here. To _really_ get this right I think is out of scope for this PR as it would brush up against other larger refactors. I could see a world where the logic of figuring out which liquid tags are restricted and if the user has authorization for each one, living somewhere else. I could also see us refactoring liquid tag documentation logic out differently as well. Maybe the liquid tag has a method to generate the documentation for itself, given a user, and it would have the power to generate the documentation based on the context it's being used in and the roles of the user.

Either way, this approach seemed like a happy middle ground. I'm open to other ideas! I also considered removing this logic altogether, always showing for the documentation for this liquid tag, and adding a footnote along the lines of:

> This liquid tag can only be used by authorized users.